cmd/tracee-rules/main.go

```diff
@@ -136,26 +136,18 @@ func main() {
 				return fmt.Errorf("constructing engine: %w", err)
 			}
 
-			if server.ShouldStart(c) {
-				httpServer := server.New(c.String(server.ListenEndpointFlag))
-
-				if c.Bool(server.MetricsEndpointFlag) {
-					err := e.Stats().RegisterPrometheus()
-					if err != nil {
-						logger.Error("registering prometheus metrics", "error", err)
-					} else {
-						httpServer.EnableMetricsEndpoint()
-					}
-				}
-
-				if c.Bool(server.HealthzEndpointFlag) {
-					httpServer.EnableHealthzEndpoint()
-				}
+			httpServer, err := server.PrepareServer(
+				c.String(server.ListenEndpointFlag),
+				c.Bool(server.MetricsEndpointFlag),
+				c.Bool(server.HealthzEndpointFlag),
+				c.Bool(server.PProfEndpointFlag),
+			)
 
-				if c.Bool(server.PProfEndpointFlag) {
-					httpServer.EnablePProfEndpoint()
-				}
+			if err != nil {
+				return err
+			}
 
+			if httpServer != nil {
```

Comment:
Should we have a log if httpServer is nil?

Reply:
hmm...why? if the user didn't pass any flag to enable it, why log?

Reply:
I got it. I was already rethinking my question, since rules are going to die.